maraboupy/MarabouCore.cpp

```diff
@@ -109,16 +110,29 @@ void addAbsConstraint(InputQuery& ipq, unsigned b, unsigned f){
 }
 
 void createInputQuery(InputQuery &inputQuery, std::string networkFilePath, std::string propertyFilePath){
-  AcasParser* acasParser = new AcasParser( String(networkFilePath) );
-  acasParser->generateQuery( inputQuery );
-  String propertyFilePathM = String(propertyFilePath);
-  if ( propertyFilePath != "" )
-    {
-      printf( "Property: %s\n", propertyFilePathM.ascii() );
-      PropertyParser().parse( propertyFilePathM, inputQuery );
-    }
-  else
-    printf( "Property: None\n" );
+  try{
+    AcasParser* acasParser = new AcasParser( String(networkFilePath) );
+    acasParser->generateQuery( inputQuery );
+
+    bool success = inputQuery.constructNetworkLevelReasoner();
+    if ( success )
+      printf("Successfully created a network level reasoner.\n");
+    else
+      printf("Warning: network level reasoner construction failed.\n");
+
+    String propertyFilePathM = String(propertyFilePath);
+    if ( propertyFilePath != "" )
+      {
+        printf( "Property: %s\n", propertyFilePathM.ascii() );
+        PropertyParser().parse( propertyFilePathM, inputQuery );
+      }
+    else
+      printf( "Property: None\n" );
+  }
+  catch(const InputParserError &e){
+        printf( "Caught an InputParserError. Code: %u. Message: %s\n", e.getCode(), e.getUserMessage() );
+        exit(1);
```

Comment:
exit(1) is a very violent way to handle an exception. How are C++ exceptions handled in other areas of the Python code?

Reply:
You are right, common practice seems to do nothing, that is, return with whatever partial results one may have; so this is what I have switched to now (there is nothing to return, but the input query may have been partly populated by this point -- which is exactly what was happening in my original example).